Assets/Scripts/Data/Main/PlayerData.cs

```diff
@@ -76,16 +79,55 @@ public class PlayerData : IPlayerData {
 		if (ReadyToLevelUp()) {
 			this._xp -= IXPForLevel;
 			this._level++;
+            LeveledUp = true;
 		}	
 		return this._level;
 	}
 
     public void LevelUpCheat() {
         this._level++;
 		this._xp = 0;
+        LeveledUp = true;
+    }
+
+    bool LeveledUp = false;
+    bool CalledWait = false;
+
+    void Window(int ID)
+    {
+        
+    }
+    void Start()
+    {
+        LeveledUp = false;
+    }
+    void Update()
+    {
+        
+        if (LeveledUp && !CalledWait)
+        {
+            OnGUI();
+            CalledWait = true;
+            StartCoroutine(Wait());
+        }
+    }
+    void OnGUI()
```

Comment:
This should conceivably work, but I don't think we should use OnGUI; it's more of a legacy function that's better left for debugging. The rest of the code runs on Unity Canvas: https://docs.unity3d.com/Manual/UICanvas.html. Ideally the level up screen should also be implemented in Canvas. Let me know if you have any questions about using it (it involves working more in the inspector and scene than in code. For what you need to work on in code, you'll need the `using UnityEngine.UI ;` statement at the top of the code.